uaclient/config.py

```diff
@@ -828,10 +828,27 @@ def write_cfg(self, config_path=None):
         cfg_dict["ua_config"] = {
             key: getattr(self, key) for key in UA_CONFIGURABLE_KEYS
         }
+
         content += yaml.dump(cfg_dict, default_flow_style=False)
         util.write_file(config_path, content)
 
 
+def get_config_path(config_path=None) -> str:
+    """Get config path to be used when loading config dict."""
+    if config_path:
+        return config_path
+
+    config_path = DEFAULT_CONFIG_FILE
+    local_cfg = os.path.join(os.getcwd(), os.path.basename(config_path))
+    if os.path.exists(local_cfg):
+        config_path = local_cfg
+
+    if os.environ.get("UA_CONFIG_FILE"):
+        config_path = os.environ.get("UA_CONFIG_FILE")
+
+    return config_path
```

Comment:
We can simplify this function a bit by:
1.  dropping the config_path param and only calling this function when we need to.
2. Make it a bit more DRY by changing the order we look at env variable since it overrides what we setup anyway.
3. returning early when we have the value we want
```suggestion
def get_config_path() -> str:
    """Get config path to be used when loading config dict."""
    if os.environ.get("UA_CONFIG_FILE"):
        return os.environ.get("UA_CONFIG_FILE")

    local_cfg = os.path.join(os.getcwd(), os.path.basename(DEFAULT_CONFIG_FILE))
    if os.path.exists(local_cfg):
        return local_cfg
    return DEFAULT_CONFIG_FILE
```